examples/hosts/app-integration/external-controller/src/controller.ts

```diff
@@ -0,0 +1,90 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { EventEmitter } from "events";
+import { IKeyValueDataObject } from "./kvpair-dataobject";
+
+/**
+ * IDiceRoller describes the public API surface for our dice roller data object.
+ */
+export interface IDiceRollerController extends EventEmitter {
+    /**
+     * Get the dice value as a number.
+     */
+    readonly value: number;
+
+    /**
+     * Roll the dice.  Will cause a "diceRolled" event to be emitted.
+     */
+    roll: () => void;
+
+    /**
+     * The diceRolled event will fire whenever someone rolls the device, either locally or remotely.
+     */
+    on(event: "diceRolled", listener: () => void): this;
+}
+
+// The data is stored in a key-value pair data object, so we'll use this key for storing the value.
+const diceValueKey = "diceValue";
+
+/**
+ * The DiceRoller is our data object that implements the IDiceRoller interface.
+ */
+export class DiceRollerController extends EventEmitter implements IDiceRollerController {
```

Comment:
my primary concern with this class is that it takes us WAY beyond the 10 line hello world we've been driving towards. Is this demo more geared towards a step2 of the tutorial? Or is there a way we can reduce the code here by pushing things upstream into the framework itself?

Reply:
I'm sure we can improve here, in particular the `initializeFromExisting` is a bit clumsy.  But we need to get it functionally working first and this is the easiest way I saw to achieve that while still achieving the external controller aspect.

Reply:
if we were able to move data initialization into the data object, this controller would be much simpler. So this certainly works for now and we can look at possibilities of removing that boilerplate later.